src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2031,6 +2072,34 @@ protected virtual void MarkMethodBody (MethodBody body)
 			PostMarkMethodBody (body);
 		}
 
+		static bool IsWorthConvertingToThrow (MethodBody body)
```

Comment:
Hmm, this optimization would be better for all ConvertToThrow receivers. Could you move it there?

Reply:
I moved this out to MethodBodyScanner and I now call from a number of places